easybuild/easyblocks/p/pytorch.py

```diff
@@ -253,7 +255,42 @@ def test_step(self):
             'python': self.python_cmd,
             'excluded_tests': ' '.join(excluded_tests)
         })
-        super(EB_PyTorch, self).test_step()
+
+        (tests_out, tests_ec) = super(EB_PyTorch, self).test_step(return_output_ec=True)
+
+        ran_tests_hits = re.findall(r"^Ran (?P<test_cnt>[0-9]+) tests in", tests_out, re.M)
+        test_cnt = 0
+        for hit in ran_tests_hits:
+            test_cnt += int(hit)
+
+        failed_tests = nub(re.findall(r"^(?P<failed_test_name>.*) failed!\s*$", tests_out, re.M))
+        failed_test_cnt = len(failed_tests)
+
+        if failed_test_cnt:
+            max_failed_tests = self.cfg['max_failed_tests']
+
+            test_or_tests = 'tests' if failed_test_cnt > 1 else 'test'
+            failed_tests_txt = '\n'.join('* %s' % t for t in sorted(failed_tests))
+            msg = "%d %s (out of %d) failed:\n%s"
+
+            if max_failed_tests == 0:
+                raise EasyBuildError(msg, failed_test_cnt, test_or_tests, test_cnt, failed_tests_txt)
+            else:
+                msg += '\n\n' + ' '.join([
+                    "The PyTorch test suite is known to include some flaky tests,",
+                    "which may fail depending on the specifics of the system or the context in which they are run.",
+                    "For this PyTorch installation, EasyBuild allows up to %d tests to fail." % max_failed_tests,
+                    "We recommend to double check that the failing tests listed above ",
+                    "are known to be flaky, or do not affect your intended usage of PyTorch.",
+                    "In case of doubt, reach out to the EasyBuild community (via GitHub, Slack, or mailing list).",
+                ])
+                print_warning(msg, failed_test_cnt, test_or_tests, test_cnt, failed_tests_txt)
```

Comment:
I find this unreadable: The immediately visible text doesn't contain placeholders for those 4(!) arguments.

So as the 4 placeholders in the initial `msg` will always be inserted, I'd strongly suggest to do it above. See the suggestion

```suggestion
                print_warning(msg)
```